### ISSUE TITLE
[Feature Request] Recurring event support for VCalendar (RFC 5545)

### ISSUE THREAD

**Issue by egeersoz** [vuetifyjs/vuetify#8950]
### Problem to solve
Most calendar apps support recurring events. Indeed, not being able to specify if an event is a one-time event or a repeating event is a considerable gap in functionality.

The prevailing standard for recurring events is section 3.3.10 of RFC 5545 (https://tools.ietf.org/html/rfc5545#section-3.3.10), which defines the specification for recurrence rules on calendar events.

It would be great if VCalendar had built-in support for this, so that the calendar can calculate and properly render recurring events based on their recurrence rule, without having to rely on a server to perform those calculations.

### Proposed solution
A new `rrule` prop on event objects, which takes a recurrence rule string or object (as per the spec) and automatically renders event instances on the calendar.

Support for the rrule.js plugin (https://github.com/jakubroztocil/rrule) may be strongly desirable, as it would eliminate the need for the Vuetify team to write their own recurrence rule parser/calculator.

FullCalendar has an example implementation: https://fullcalendar.io/docs/rrule-plugin

**Comment by MajesticPotatoe** [comment]
We shy away from external package dependencies for component functionality, though if its to much of an undertaking to incorporate it internally, we usually have no problem providing ways to hook into external packages (eg: vue-the-mask and vee-validate).

Agree though having recurring events would be a nice addition.

**Comment by egeersoz** [comment]
Yeah, the spec is fairly complex, which is why FullCalendar relies on rrule.js to do it (although it does offer some very basic, i.e. daily and weekly, recurrence functionality out of the box).

**Comment by ClickerMonkey** [comment]
I'm hesitant to reference a 3rd party library in this instance, but I think we could create an example which shows how to use rrule.js and `v-calendar` together. There are many ways people are feeding events into the calendar and building in support to rrule.js now might set an unwanted precedent. What does @johnleider  think?

**Comment by stephane303** [comment]
Dead end ?

**Comment by shelbyecook** [comment]
Any luck?

**Comment by mhmdali98** [comment]
Did you add this feature?

**Comment by kazimKoc** [comment]
Any update ?

**Comment by egeersoz** [comment]
Hi Vuetify team,

I know you're all busy with Vuetify 3, but wanted to check on the status of this one. As @ClickerMonkey said, even a working example that uses an external library like rrule.js would go a long way towards expanding use cases for `v-calendar`.

**Comment by johnleider** [comment]
> I'm hesitant to reference a 3rd party library in this instance, but I think we could create an example which shows how to use rrule.js and `v-calendar` together. There are many ways people are feeding events into the calendar and building in support to rrule.js now might set an unwanted precedent. What does @johnleider think?

No 3rd party libraries. We can **integrate** with them, but they cannot be a requirement.

**Comment by egeersoz** [comment]
Hello. Since `v-calendar` is being worked on for v3, I wanted to bring up this request again. Being able to integrate with rrule.js would extend the usefulness of this component a lot.
